src/main/java/hudson/plugins/sonar/utils/SonarUtils.java

```diff
@@ -235,7 +236,9 @@ public static StringCredentials getCredentials(Run<?, ?> build, String credentia
 
   public static String getTokenProperty(SonarInstallation inst, HttpClient client) {
     try {
-      if (getVersion(inst, client).compareTo(new Version("10.0")) < 0) {
+      if (getVersion(inst, client).compareTo(new Version("10.0")) < 0
+          // SonarCloud always returns version 8.X.X. Then we have to check the url to know if it is SonarCloud or not.
+          && !isSonarCloud(inst)) {
```

Comment:
Could you move this condition before the other one? It would prevent making a potentially useless call to the instance to get the version.
Also, you can remove the comment as I believe the code is self-explanatory.

Reply:
You are right, I have already applied the modifications you asked for. ;-)